python/nwis_client/src/hydrotools/nwis_client/cli.py

```diff
@@ -0,0 +1,92 @@
+import click
+from hydrotools.nwis_client import IVDataService
+from hydrotools.nwis_client import _version as CLIENT_VERSION
+from typing import Tuple
+from pathlib import Path
+import pandas as pd
+
+class TimestampParamType(click.ParamType):
+    name = "timestamp"
+
+    def convert(self, value, param, ctx):
+        if isinstance(value, pd.Timestamp):
+            return value
+
+        try:
+            return pd.Timestamp(value)
+        except ValueError:
+            self.fail(f"{value!r} is not a valid timestamp", param, ctx)
+
+def write_to_csv(
```

Comment:
I imagine this would just be another method that wraps `pandas.DataFrame.to_json`. Opportunities to generalize the output in the future. Perhaps when we hash out the logging details.